File: 179-largest-number/largest-number.cpp
class Solution {
public:
    string largestNumber(vector<int>& nums) {
        std::vector<std::string> strNums;
        for (int num : nums) {
            strNums.push_back(std::to_string(num));
        }
        std::sort(strNums.begin(), strNums.end(), [](const std::string& a, const std::string& b) {
            return a + b > b + a;
        });
        if (strNums[0] == "0") {
            return "0";
        }
        std::string result;
        for (const std::string& str : strNums) {
            result += str;
        }
        return result;
    }
};

File: 179-largest-number/largest-number.py
class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        snums = list(map(str, nums))
        print(snums)
        snums.sort(key=lambda x: x*10, reverse = True)
        if snums[0] == "0":
            return "0"
        return ''.join(snums)